### ISSUE TITLE
planned activation functions for gradient descent

### ISSUE THREAD

**Issue by Kiyoshika** [Kiyoshika/Generalized-Model-Framework#2]
planning the following additional activation functions for gradient descent:
* ReLU (non-negative regression)
* softmax (to handle multi-class classifiers)
* exponential activations (undetermined for now, will need a bit of experimenting)

**Comment by Kiyoshika** [comment]
implemented ReLU in 8cb48fa0974320b1c840eb28c7de25b96053952b

thought about adding softmax, but would require adding support for vector-based targets. instead, I will (later) think about how to support multi-classification for label-encoded binary vars (maybe a min-distance to value type thing)
